server/build.gradle

```diff
@@ -21,6 +21,7 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     testImplementation 'junit:junit:4.13.1'
+    compileOnly 'androidx.annotation:annotation:1.3.0' // 1.3.0 doesn't depend on Kotlin stdlib.
```

Comment:
Nope, the server should be buildable without any java/kotlin dependencies (and without gradle): [`build_without_gradle.sh`](https://github.com/Genymobile/scrcpy/blob/master/server/build_without_gradle.sh)

(Btw, I just removed the `implementation fileTree` line above: 7f2f5950f2624c1a3fd847af6248bcbef71a3124)

Reply:
Hmm, is it OK to download a prebuilt into a directory (e.g. `prebuilt-compileOnly`) and also use everything in it as classpath for `javac`? That should work both ways. I really want the IDE to help me not call API 19 stuff on API 21.

> and without gradle

Any other reason than using build system within build system is weird? (And the flex of knowing how to manually produce an APK? 😁 )

Reply:
> is it OK to download a prebuilt into a directory

Nope. If it was a necessary dependency, we could discuss what to do, but here they are just annotations, it does not worth the dependency.

> Any other reason than using build system within build system is weird?

It is used to build the official Debian package: https://salsa.debian.org/yangfl-guest/scrcpy/-/blob/ab5d09e733bb9635b4410693ceff95dbc859dd4c/debian/rules#L41
There were problems with gradle/kotlin (Debian provides its own packages for android tools).